common/src/main/java/io/netty/util/concurrent/LimitLatch.java

```diff
@@ -0,0 +1,111 @@
+package io.netty.util.concurrent;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.AbstractQueuedSynchronizer;
+
+/**
+ * Shared latch that allows the latch to be acquired a limited number of times
+ * after which all subsequent requests to acquire the latch will fail.
+ */
+public class LimitLatch {
+
+    private class Sync extends AbstractQueuedSynchronizer {

```

Comment:
this is used to solve "busy loop" problem. 
1. because there are incoming in the backlog queue, select() would block, so the loop would keep spinning. this blocking method is meant to slove this "busy loop" problem.
2. i made the assumption that we won't run task in boss loop.
3. another way to solve "busy loop" problem is we return a special value from io.netty.channel.nio.AbstractNioChannel.NioUnsafe#read() to indicate that we have read too much, so the loop may choose to pause a little while if there is no task to run. java.lang.Integer#MAX_VALUE would be perfect for this.